ui/src/components/directory/topbar/eclSearch/EclTree.vue

```diff
@@ -0,0 +1,397 @@
+import { useToast } from "primevue/usetoast";
+
+const {
+  root,
+  expandedKeys,
+  selectedKeys,
+  createLoadMoreNode,
+  createTreeNode,
+  onNodeCollapse,
+  onNodeDblClick,
+  onNodeExpand,
+  onRowClick,
+  loadMore,
+  selectAndExpand,
+  scrollToHighlighted,
+  loadMoreChildren,
+  locateChildInLoadMore,
+  selectKey,
+  selectedNode
+} = setupTree();
+
+const toast = useToast();
+
+const dialogRef = inject("dialogRef");
+
+const rootConceptIri = ref("");
+const conceptAggregates: Ref<ConceptAggregate[]> = ref([]);
+const hoveredResult: Ref<ConceptSummary> = ref({} as ConceptSummary);
+const overlayLocation: Ref = ref({});
+const loading = ref(false);
+const totalCount = ref(0);
+const superiorsCount = ref(0);
+const pageSize = ref(20);
+
+const altTreeOP = ref();
+
+watch(
+  () => dialogRef.value.data.focus.iri,
+  async newValue => {
+    selectedKeys.value = {};
+    expandedKeys.value = {};
+    if (newValue) {
+      await getConceptAggregates(newValue);
+      for (const aggregate of conceptAggregates.value) {
+        await createTree(aggregate.concept, aggregate.children);
+      }
+    }
+  }
+);
+
+watch(loading, newValue => {
+  if (newValue) hidePopup(overlayLocation.value);
+});
+
+onMounted(async () => {
+  setRootConceptIri();
+  if (rootConceptIri.value) {
+    await getConceptAggregates(rootConceptIri.value);
+    for (const aggregate of conceptAggregates.value) {
+      await createTree(aggregate.concept, aggregate.children);
+    }
+    if (root.value.length < superiorsCount.value) {
+      root.value.push(
+        createLoadMoreNode(
+          createTreeNode(dialogRef.value.data.focus.iri, dialogRef.value.data.focus.name, [{ "@id": IM.CONCEPT, name: "Concept" }], false, null),
+          1,
+          superiorsCount.value
+        )
+      );
+    }
+    if (dialogRef.value.data.currentValue) {
+      await findSelected(dialogRef.value.data.currentValue.iri);
+    }
+  }
+});
+
+onBeforeUnmount(() => {
+  if (isObject(overlayLocation.value) && isArrayHasLength(Object.keys(overlayLocation.value))) {
+    hidePopup(overlayLocation.value);
+  }
+  dialogRef.value.close();
+});
+
+function setRootConceptIri() {
+  if (dialogRef.value && isObjectHasKeys(dialogRef.value, ["data"]) && isObjectHasKeys(dialogRef.value.data, ["focus"]))
+    rootConceptIri.value = dialogRef.value.data.focus.iri;
+}
+
+async function getConceptAggregates(iri: string): Promise<void> {
+  loading.value = true;
+  let superiors = [];
+  if (dialogRef.value.data.type === "property") {
+    const results = await EntityService.getSuperiorPropertiesPaged(rootConceptIri.value, 1, pageSize.value);
+    superiors = results.result;
+    superiorsCount.value = results.totalCount;
+  } else {
+    const results = await EntityService.getSuperiorPropertyValuesPaged(rootConceptIri.value, 1, pageSize.value);
+    superiors = results.result;
+    superiorsCount.value = results.totalCount;
+  }
+  for (const superior of superiors) {
```

Comment:
i think this is the bit that is making the ui slow 
- one option is to look at the Promise.all()
- another is to create a custom call at the backend for all the data that is needed
- or do not get more than one level in one go (let the user expand each one and then make the call for each children) but with an indicator whether it has children or not (another ask sparql?)